### PR TITLE
parser: clean up unused code

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -1244,9 +1237,6 @@ CreateIndexStmt:
                 	Table: $6.(*ast.TableName),
 			IndexColNames: $8.([]*ast.IndexColName),
 		}
-		if yylex.(*lexer).root {
```
Comment (shenli) [inline parser/parser.y]: Why remove this?

File: parser/scanner.l [hunk 2]
```diff
@@ -85,30 +82,6 @@ func (l *lexer) Expr() ast.ExprNode {
...
-
-func (l *lexer) SetInj(inj int) {
-	l.inj = inj
-}
-
-func (l *lexer) SetPrepare() {
-	l.prepare = true
-}
-
-func (l *lexer) IsPrepare() bool {
```
Comment (shenli) [inline parser/scanner.l]: Only prepare statement allows placeholder. So how can we check if a placeholder if valid without this?

**Comment by shenli** [comment]
Is this PR for performance consideration?

**Comment by shenli** [comment]
LGTM
@coocood PTAL

**Comment by tiancaiamao** [comment]
this PR is not for performance, just clean up dead code.
maybe we'll replace the generated lexer with a handwritten one,  before that we need do some clean up. 

currently, the code is bit mess:
1. there are many `xxx.(*lexer)` assertion in parser.y, means parser dependent on specific implementation rather than the standard `yyLexer` interface, that's a bad smell.

2. to provide better error information, lexer and parser's context don't have a clean boundary. 
because the input maybe conform to grammar, but not conform to legal semantics... parser know it's an error but don't have the line and col information, so it ask lexer for that.

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**